packages/flutter_tools/lib/src/build_system/targets/android.dart

```diff
@@ -309,3 +333,60 @@ const Target androidx64ProfileBundle = AndroidAotBundle(androidx64Profile);
 const Target androidArmReleaseBundle = AndroidAotBundle(androidArmRelease);
 const Target androidArm64ReleaseBundle = AndroidAotBundle(androidArm64Release);
 const Target androidx64ReleaseBundle = AndroidAotBundle(androidx64Release);
+
+// Rule that performs setup verification for deferred components
+// and copies files to the intermediate dirs.
+class AndroidAotDeferredComponentsBundle extends Target {
+  /// Create an [AndroidAotDeferredComponentsBundle] implementation for a given [targetPlatform] and [buildMode].
+  const AndroidAotDeferredComponentsBundle(this.dependency);
+
+  /// The [AndroidAot] instance this bundle rule depends on.
+  final AndroidAotBundle dependency;
+
+  /// The name of the produced Android ABI.
+  String get _androidAbiName {
+    return getNameForAndroidArch(
+      getAndroidArchForName(getNameForTargetPlatform(dependency.targetPlatform)));
+  }
+
+  @override
+  String get name => 'android_aot_deferred_components_bundle_${getNameForBuildMode(dependency.buildMode)}_'
+    '${getNameForTargetPlatform(dependency.targetPlatform)}';
+
+  @override
+  List<Source> get inputs => <Source>[
+    // Tracking app.so is enough to invalidate the dynamically named
+    // loading unit libs as changes to loading units guarantee
+    // changes to app.so as well. This task does not actually
+    // copy app.so.
+    Source.pattern('{OUTPUT_DIR}/$_androidAbiName/app.so'),
+    Source.pattern('{OUTPUT_DIR}/$_androidAbiName/manifest.json'),
+  ];
+
+  @override
+  List<Source> get outputs => const <Source>[];
```

Comment:
you need either an output or a stampfile to keep this from re-running.

Reply:
Looks like you didn't address this?